### ISSUE TITLE
Update /usr/bin shebangs to Python3?

### ISSUE THREAD

**Issue by RollingStar** [beetbox/beets#4604]
### Problem

YouTuber found error because beets calls for Python3 but actually requires python36+.

https://youtu.be/_-WeZhlU0_I?t=492

Probably this code needs to change:

https://github.com/beetbox/beets/search?q=%2Fusr%2Fbin

**Open question**: Specify Python36 in the shebangs? I bet this would cause compatibility issues with shells that only have generic python3 installed with no subversion number (3.**6**).

### Setup

* OS: Linux

**Comment by JOJ0** [comment]
I'll have to rewatch the video at home, but the first thing that comes up is: He's installing beets 1.4.9. Video is from 1 year ago. Why? Wasn't anything newer out this time? I think 1.6.0 even...will check! He doesnt install a distro package but directly via pip3

**Comment by wisp3rwind** [comment]
> [](https://github.com/beetbox/beets/search?q=%2Fusr%2Fbin)
> 
> **Open question**: Specify Python36 in the shebangs? I bet this would cause compatibility issues with shells that only have generic python3 installed with no subversion number (3.**6**).

No. In particular, manually making these lines specific to minor Python versions will fail with any other supported python which is not 3.6 (i.e. 3.7). We could change it to `python3`.

However, I think none of the scripts that this search turns up are actually relevant to users: Almost all of them are developer tools, except for the `beet` "executable". However, even that isn't actually used as far as I can tell (In fact, could we even remove it?), since we install the script via `entry_points` in `setup.py`, which generates a stub file that should refer to the correct python version which has beets installed (check `cat $(which beet)` on your system).

I haven't watched the video, but that sounds more like a botched-up Python/beets installation.

**Comment by sampsyo** [comment]
> However, even that isn't actually used as far as I can tell (In fact, could we even remove it?), since we install the script via `entry_points` in `setup.py`, which generates a stub file that should refer to the correct python version which has beets installed (check `cat $(which beet)` on your system).

All correct! It might be a good idea to remove this… it has been vaguely nice for testing without installing in the past, but even I don't do that anymore (`pip install -e` is a much better option).

**Comment by RollingStar** [comment]
Sounds like the issue is:

1. Remove /usr/bin/python shebangs from code that no longer needs it (/beet)
2. Update shebangs to python3 for remaining code. Even if it's dev/test code, that's important for devs and testing.
3. The issue in the OP is apparently solved already.

> /usr/bin/env python
> 
> usually is the correct way, thus everything your github code search link shows seems to be correct. Mostly things not handled by setuptools are found.

Isn't explicit better than implicit? python (no 3) is the PEP way to do it, but some [disagree](https://discuss.python.org/t/pep-394-clarify-bullet-for-usr-bin-env-python/7953).

https://peps.python.org/pep-0394/

> For scripts that are only expected to be run in an activated virtual environment, shebang lines can be written as #!/usr/bin/env python, as this instructs the script to respect the active virtual environment.

**Comment by JOJ0** [comment]
I'm for removing beets executable.

Anything else to do here @RollingStar you wann take over here and state the todos? Thanks :-))

**Comment by RollingStar** [comment]
I don't think I understand it well enough to state the todos.

**Comment by sampsyo** [comment]
Here's what a potential PR should probably do:

* Remove the included `beet` executable.
* Grep for `bin\/env python` and change all other random scripts we have lying around (there are a few) to use `python3` instead of `python`.

**Comment by bal-e** [comment]
I had a look over PEP 394, and I agree that changing `env python` to `env python3` makes sense.  I didn't realize that the `beet` script in the repo simply goes unused; if that should be removed, I'm happy to do it.  I'll open a (single) PR for both of these.
